### PR TITLE
Issue #7657: Update doc for ReturnCount

### PR DESCRIPTION
**Pull request by kexuancoco**
Issue #7657 :Update doc for ReturnCount
ex1
ex2
ex3
ex4
Output of the configuration(it doesn't allow more than three return statements per method): 
```
$ type src\com\test\config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="ReturnCount">
            <property name="max" value="3"/>
        </module>
    </module>
</module>

$ type src\com\test\Test.java 
package com.test;

public class Test {
    public int test1(int a){
        if(a == 0) return 1;
        return 0;
    }//OK
    public int test2(int a){
        if(a < 0) return -1;
        if(a == 0) return 1;
        return 0;
    }//OK
    public int test3(int a){
        if(a < -2) return -2;
        if(a == 0) return 0;
        if(a > 2) return 2;
        return 1;
    }//Violation,more than three return statements
}

$ java -jar checkstyle-8.39-all.jar -c src/com/test/config.xml src/com/test/Test.java
Picked up JAVA_TOOL_OPTIONS: -Duser.language=en
Starting audit...
[ERROR] D:\projects\MyClass\src\com\test\Test.java:13:5: Return count is 4 (max allowed for non-void methods/lambdas is 3). [ReturnCount]
Audit done.
Checkstyle ends with 1 errors.
```
Output of the configuration(it doesn't allow any return statements per void method):
```
$ type src\com\test\config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="ReturnCount">
            <property name="maxForVoid" value="0"/>
        </module>
    </module>
</module>

$ type src\com\test\Test.java
package com.test;

public class Test {
    public void test1(){ } //OK
    public void test2(){ return; } //Violation
}


$ java -jar checkstyle-8.39-all.jar -c src/com/test/config.xml src/com/test/Test.java
Picked up JAVA_TOOL_OPTIONS: -Duser.language=en
Starting audit...
[ERROR] D:\projects\MyClass\src\com\test\Test2.java:5:5: Return count is 1 (max allowed for void methods/constructors/lambdas is 0). [ReturnCount]
Audit done.
Checkstyle ends with 1 errors.
```
Output of the configuration(doesn't allow more than 2 return statements per method and more than 1 return statements per void method):
```
$ type src\com\test\config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="ReturnCount">
            <property name="max" value="2"/>
            <property name="maxForVoid" value="1"/>
        </module>
    </module>
</module>

$ type src\com\test\Test.java
package com.test;

public class Test {
    public void test1(){ } //OK
    public void test2(){ return; } //OK
    public void test3(int a){
        if(a == 0) return;
        return;
    }//Violation,more than 1 return statements
    public int test4(int a){
        if(a < 0) return 0;
        return 1;
    }//OK
    public int test5(int a){
        if(a > 0) return -1;
        if(a == 0) return 0;
        return 1;
    }//Violation,more than 2 return statements
}

$ java -jar checkstyle-8.39-all.jar -c src/com/test/config.xml src/com/test/Test.java
Picked up JAVA_TOOL_OPTIONS: -Duser.language=en
Starting audit...
[ERROR] D:\projects\MyClass\src\com\test\Test.java:6:5: Return count is 2 (max allowed for void methods/constructors/lambdas is 1). [ReturnCount]
[ERROR] D:\projects\MyClass\src\com\test\Test.java:14:5: Return count is 3 (max allowed for non-void methods/lambdas is 2). [ReturnCount]
Audit done.
Checkstyle ends with 2 errors.
```
Output of the configuration(it doesn't allow more than three return statements per method for all methods):
```
$type src\com\test\config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="ReturnCount">
            <property name="max" value="3"/>
            <property name="format" value="^$"/>
        </module>
    </module>
</module>

$type src\com\test\Test.java
package com.test;

public class Test {
    public int test1(int a){
        if(a < 0) return -1;
        if(a == 0) return 0;
        return 1;
    }//OK
    public int test2(int a){
        if(a < -2) return -2;
        if(a == 0) return 0;
        if(a > 2) return 2;
        return 1;
    }//Violation,don't allow 4 return statements
}

$ java -jar checkstyle-8.39-all.jar -c src/com/test/config.xml src/com/test/Test.java
Picked up JAVA_TOOL_OPTIONS: -Duser.language=en
Starting audit...
[ERROR] D:\projects\MyClass\src\com\test\Test.java:9:5: Return count is 4 (max allowed for non-void methods/lambdas is 3). [ReturnCount]
Audit done.
Checkstyle ends with 1 errors.
```
Output of configuration( it doesn't allow any return statements in constructors, more than one return statement in all lambda expressions and more than two return statements in methods) 
```
$ type src\com\test\config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="ReturnCount">
            <property name="max" value="0"/>
            <property name="tokens" value="CTOR_DEF"/>
        </module>
        <module name="ReturnCount">
            <property name="max" value="1"/>
            <property name="tokens" value="LAMBDA"/>
        </module>
        <module name="ReturnCount">
            <property name="max" value="2"/>
            <property name="tokens" value="METHOD_DEF"/>
        </module>
    </module>
</module>

$ type src\com\test\Test.java
package com.test;

public class Test{
    public class test1 {
            public test1() {
            }
    }//OK

    public class test2{
        public test2(){ return;}
    }//Violation,don't allow return statements in constructors

    public int test3(int a){
        if(a < 0) return 0;
        return 1;
    }//OK

    public int test4(int a){
        if(a < 0) return -1;
        if(a > 0) return 1;
        return 0;
    }//Violation

    Operation test5  = (int x,int y) -> x + y; //OK
    Operation test6 = (int x,int y) -> { return x + y;};//OK
    Operation test7 = (int x,int y) -> {
        if(x > y) return x - y;
        return x + y;
    };//Violation,2 return statements in lambda expressions

    interface Operation {
        int operation(int x, int y);
    }
}

$ java -jar checkstyle-8.39-all.jar -c src/com/test/config.xml src/com/test/Test.java
Picked up JAVA_TOOL_OPTIONS: -Duser.language=en
Starting audit...
[ERROR] D:\projects\MyClass\src\com\test\Test.java:18:5: Return count is 3 (max allowed for non-void methods/lambdas is 2). [ReturnCount]
[ERROR] D:\projects\MyClass\src\com\test\Test.java:26:37: Return count is 2 (max allowed for non-void methods/lambdas is 1). [ReturnCount]
Audit done.
Checkstyle ends with 2 errors.
```
It seems that this configuration yields unexpected results: Checkstyle ends with only 2 errors,allow the return statements in contructors.I cannot understand what should be the problem here. Sorry for that :-(

### CODE REVIEW

**Comment by kexuancoco** [comment]
?? I make sure that command `mvn verify`  pass without violations then pull request....Maybe I changed something accidentally,sorry for that,I'll update the PR.

**Comment by strkkk** [comment]
@kexuancoco you need to escape '>' chars

Please also move violation comments on line where they are reported (method signature, not closing curly brace)
violation comments should be like `// violation` and ok comments `// OK`

**Comment by kexuancoco** [comment]
> @kexuancoco you need to escape '>' chars
> 
> Please also move violation comments on line where they are reported (method signature, not closing curly brace)
> violation comments should be like `// violation` and ok comments `// OK`

OK,thanks for your guidance!!! I will change the location and the format of comments.And could you please explain that what  "escape '>' chars" means ? 
Sorry for asking such a stupid question...I will try very hard to refrain from asking stupid questions again :-(

**Comment by strkkk** [comment]
> could you please explain that what "escape '>' chars" means ?

replace it with &gt;

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/ReturnCountCheck.java [hunk 1]
```diff
@@ -98,6 +118,13 @@
  * &lt;/module&gt;
  * </pre>
  * <p>
+ * Example for above configuration:
+ * </p>
+ * <pre>
+ * public void test1(){ } //OK
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/ReturnCountCheck.java:124]: https://github.com/checkstyle/checkstyle/wiki/Good-practice-on-how-to-update-examples please read "ATTENTION" paragraph, java code example should be the same for all configs , only trailing comments are changing.

example https://github.com/checkstyle/checkstyle/commit/418fdb1a691a4241eb93d1282637b16f184d58e6

**Review by romani — CHANGES_REQUESTED** [review]
item:

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/ReturnCountCheck.java [hunk 2]
```diff
@@ -137,6 +200,32 @@
  * &lt;/module&gt;
  * </pre>
```
Comment (aimiebell) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/ReturnCountCheck.java:201]: I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Class names should start with an upper case letter (NM_CLASS_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-class-names-should-start-with-an-upper-case-letter-nm-class-naming-convention).
Class names should be nouns, in mixed case with the first letter of each internal word capitalized. Try to keep your class names simple and descriptive. Use whole words-avoid acronyms and abbreviations (unless the abbreviation is much more widely used than the long form, such as URL or HTML).

**Comment by romani** [comment]
@kexuancoco , please fix CI violations and adddress my points

**Comment by kexuancoco** [comment]
@romani  Sorry for the late reply!Actually I don't have enough time to do this right now, but I'm fixing these issues as fast as I can! But it might take some time. May I apply to close this PR? When I have finished fixing these issues, I will create a new PR!

**Comment by romani** [comment]
Thanks for reply, lets keep PR open for month, if you did not find time for it I will close PR and can reopen it when you got a time

**Pull request closed without merging**